### PR TITLE
[AIRFLOW-7013] Automated check if Breeze image needs to be pulled

### PR DESCRIPTION
**Pull request by potiuk**
---
Issue link: [AIRFLOW-7013](https://issues.apache.org/jira/browse/AIRFLOW-7013)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by potiuk** [comment]
This one implements automated check if it is worth to pull image rather than build it locally. When the image is changed a lot (for example with buster) it makes very little sense to rebuild the image, it's better to pull it. 

This change will automatically detect if it is worth to build or pull&build image when you reach the "need rebuild phase".

**Review by feluelle — APPROVED** [review]

**Review by ashb — COMMENTED** [review]
Can you add some docs/comments in the PR somewhere about

1. what this is doing -- I'm having trouble following the logic; and
2. why (your comment in the PR should go in the script somewhere)

File: scripts/ci/_utils.sh [hunk 1]
```diff
@@ -552,6 +561,85 @@ function set_current_image_variables {
...
+
+function get_local_image_info() {
+    TMP_MANIFEST_LOCAL_JSON=$(mktemp)
+    TMP_MANIFEST_LOCAL_SHA=$(mktemp)
+    set +e
+    verbose_docker rm --force "local-airflow-manifest"  >/dev/null 2>&1
+    if ! verbose_docker create --name "local-airflow-manifest" \
+        "${AIRFLOW_LOCAL_MANIFEST_IMAGE}"  >/dev/null 2>&1 ; then
+        echo
+        echo "Local docker not available"
```
Comment (ashb) [inline scripts/ci/_utils.sh]:
```suggestion
        echo "Local airflow CI docker image not available"
```

Comment (ashb) [reply scripts/ci/_utils.sh]: Or similar - "image" being the key bit to add here.

Comment (potiuk) [reply scripts/ci/_utils.sh]: > Can you add some docs/comments in the PR somewhere about
> 
> 1. what this is doing -- I'm having trouble following the logic; and
> 2. why (your comment in the PR should go in the script somewhere)

Good point. I added comments describing both.

Comment (potiuk) [reply scripts/ci/_utils.sh]: Added full description + a bit nicer output showing pull progress as well as step of the docker build during pre-commit.

**Comment by potiuk** [comment]
All should be good now!

**Comment by potiuk** [comment]
Hey @ashb -> are you OK with those comments ?

**Pull request merged**